__tests__/endpoints.test.js

```diff
@@ -0,0 +1,395 @@
+            expect(body).toMatchObject({ status: 400, msg: 'Bad Request' })
+        }) 
+    })
+
+})
+
+describe('For adding a comment to a specified article', () => {
+    test('Status 201: inserts a new comment for the specified article and responds with the posted comment',()=>{
+        return request(app)
+        .post('/api/articles/3/comments')
+        .send({ username: "butter_bridge",
+                body: "I am northcoder student"
+        })
+        .expect(201)
+        .then(({body}) => {
+            const comment = body[0]
+            expect(comment).toMatchObject({
+                comment_id: expect.any(Number),
+                body: 'I am northcoder student',
+                article_id: 3,
+                author: "butter_bridge",
+                votes: expect.any(Number),
+                created_at: expect.any(String)
+            })
+        });
+    })
+    test('Status 404: cannot post and displays 404 error message because no such article to post the comment to',()=>{
+        return request(app)
+        .post('/api/articles/999/comments')
+        .send({ username: "daniel",
+                body: "I am northcoder student"
+        })
+        .expect(404)
+        .then(({body}) => {
+            expect(body).toMatchObject({ status: 404, msg: 'Not found' })
+        });
+    })
+
+    test('Status 401: unauthorised user tried to add a comment to an article',()=>{
+        return request(app)
+        .post('/api/articles/3/comments')
+        .send({ username: "daniel",
+                body: "I am northcoder student"
+        })
+        .expect(401)
+        .then(({body}) => {
+            expect(body).toMatchObject({ status: 401, msg: 'Unauthorized' })
+        });
+    })
+
+
+    
+})
+
+
+
+describe('Patches the article by updating the number of votes', () => {
+    test('Status 201: succesfully patches the article when a valid article is chosen and updates the number of votes when given amount to update them',()=>{
+        return request(app)
+        .patch('/api/articles/3')
+        .send({ inc_votes: 50 })
+        .expect(201)
+        .then(({body}) => {
+            expect(body).toMatchObject({
+                article_id: 3,
+                title: 'Eight pug gifs that remind me of mitch',
+                topic: 'mitch',
+                author: 'icellusedkars',
+                body: 'some gifs',
+                created_at: '2020-11-03T09:12:00.000Z',
+                votes: 50,
+                article_img_url: 'https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700'})
+        });
+    })
+
+    test('Status 404: tries to patch a non existant article and returns an error and status code',()=>{
+        return request(app)
+        .patch('/api/articles/999')
+        .send({ inc_votes: 50 })
+        .expect(404)
+        .then(({body}) => {
+            expect(body).toMatchObject({ status: 404, msg: 'Not found' })
+        })
+    })
```

Comment:
Nice tests, but could do with a couple more.

- What id the article id is not a number?
- What if the inc_votes object is empty?